### ISSUE TITLE
How to have dynamic number of dots?

### ISSUE THREAD

**Issue by nicefellow1234** [OwlCarousel2/OwlCarousel2#1881]
For instance I want to have 10 dots when I am at 1366 resolution and then when I am at 300 to 400 resolution so I only want to 2 to 3 dots to show for navigation..!

But now at full resolution I get to see minimum dots but when the resolution comes to the mobile so all of the dots start showing in there which looks very ugly as : 

![image](https://cloud.githubusercontent.com/assets/10282608/25975769/bdb465b2-36c9-11e7-848e-8dab7d9ecf89.png)

While here in full resolution it looks okay : 

![image](https://cloud.githubusercontent.com/assets/10282608/25975781/cf194c0a-36c9-11e7-8262-701b036bd1b1.png)

**Comment by Kylescottw** [comment]
Hello,
A simple fix is to edit your css.
Add a display: none property to the owl-dots class.

**Comment by nicefellow1234** [comment]
@Kylescottw  : Yes that would be but that would really make every dot invisible which I really do not want.I just want to have minimal dots in the mobile resolution so any possible way to do that?

**Comment by keefyhub** [comment]
you could look at using the `dotsEach` and `responsive` options - https://owlcarousel2.github.io/OwlCarousel2/docs/api-options.html#dotseach

```
$(".owl-carousel").owlCarousel({
  dots: true,
  loop: true,
  margin: 10,
  nav: false,
  responsive: {
    0: {
      dotsEach: 5,
      items: 1
    },
    600: {
      dotsEach: 3,
      items: 1
    },
    1200: {
      dotsEach: 1,
      items: 1
    }
  }
});
```

The problem with `dotsEach` is that it then skips slides. But if you have navigation arrows then it might not matter.

Working pen - https://codepen.io/keefyboooo/pen/mmLdXO

**Comment by nicefellow1234** [comment]
@keefyhub  : That is really great but for that as you said I will need navigation arrows so is there any way to add arrows too to the navigation dots on each side for next and previous?

**Comment by keefyhub** [comment]
You could do a carousel just for the dots, but only activate it at a specific screen size.

Im not sure if you can do it with the generated carousel dots, you may have to make your own and use [`dotsContainer`](https://owlcarousel2.github.io/OwlCarousel2/docs/api-options.html#dotscontainer)

**Comment by gisfrancis** [comment]
> you could look at using the `dotsEach` and `responsive` options - https://owlcarousel2.github.io/OwlCarousel2/docs/api-options.html#dotseach
> 
> ```
> $(".owl-carousel").owlCarousel({
>   dots: true,
>   loop: true,
>   margin: 10,
>   nav: false,
>   responsive: {
>     0: {
>       dotsEach: 5,
>       items: 1
>     },
>     600: {
>       dotsEach: 3,
>       items: 1
>     },
>     1200: {
>       dotsEach: 1,
>       items: 1
>     }
>   }
> });
> ```
> 
> The problem with `dotsEach` is that it then skips slides. But if you have navigation arrows then it might not matter.
> 
> Working pen - https://codepen.io/keefyboooo/pen/mmLdXO

its usefull

**Comment by gustavohcmelo** [comment]
I dont know if you already can did it. But that is my solution:

```
var total_itens = Math.ceil(document.getElementsByClassName('item-owl').length / 5)

$(".owl-two").owlCarousel({
            margin: 20,
            responsiveClass:true,
            dotsEach: total_itens,
            responsive:{
                0:{
                    items:1,
                    nav:false
                },
                600:{
                    items:3,
                    nav:false
                },
                1600:{
                    items:4,
                    nav:false
                },
            }
        });
```
in this case, there will always be 5 dots. There are other validations to be done. But it worked for me.
